dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/teradata/TeradataLogsConnector.java

```diff
@@ -434,15 +442,20 @@ public void addTasksTo(List<? super Task<?>> out, @Nonnull ConnectorArguments ar
         LOG.info("Exporting query log for " + intervals);
         SharedState state = new SharedState();
         for (ZonedInterval interval : intervals) {
-            String file = ZIP_ENTRY_PREFIX + DateTimeFormatter.ISO_OFFSET_DATE_TIME.format(interval.getStartUTC()) + ".csv";
+            String file = ZIP_ENTRY_PREFIX + DateTimeFormatter.ISO_OFFSET_DATE_TIME.format(
+                interval.getStartUTC()) + ".csv";
+            Consumer<Long> consumer = count -> updateMetadata(ZIP_ENTRY_PREFIX, count, interval);
```

Comment:
I would suggest to create a method `createMetadataUpdater` and use it in all the places where updateMetadata is used in the lambda currently.

```
Consumer<Long> createMetadataUpdater(String filePrefix, ZonedInterval interval) { ... }
```